pkg/webui/console/containers/webhook-add/webhook-add.js

```diff
@@ -0,0 +1,192 @@
+    createWebhook: PropTypes.func.isRequired,
+    getWebhook: PropTypes.func.isRequired,
+    navigateToList: PropTypes.func.isRequired,
+    templateId: PropTypes.string.isRequired,
+    webhookTemplate: PropTypes.webhookTemplate,
+  }
+
+  static defaultProps = {
+    webhookTemplate: undefined,
+  }
+
+  modalResolve = () => null
+  modalReject = () => null
+
+  state = {
+    error: undefined,
+    displayOverwriteModal: false,
+    existingId: undefined,
+  }
+
+  @bind
+  async convertTemplateToWebhook(values) {
+    const { webhookTemplate: template, appId, createApplicationApiKey } = this.props
+    const { webhook_id, ...fields } = values
+
+    const headers = Object.keys(template.headers || {}).reduce((acc, key) => {
+      const val = template.headers[key]
+      const headerValue = val.replace(/{([a-z0-9_-]+)}/i, (_, field) => values[field])
+      if (headerValue !== '') {
+        acc[key] = headerValue
+      }
+      return acc
+    }, {})
+
+    const webhook = {
+      ids: {
+        webhook_id: values.webhook_id,
+      },
+      template_ids: template.ids,
+      format: template.format,
+      headers,
+      template_fields: fields,
+      base_url: urlTemplate.parse(template.base_url).expand(fields),
+      uplink_message: pathExpand(template.uplink_message, fields),
+      join_accept: pathExpand(template.join_accept, fields),
+      downlink_ack: pathExpand(template.downlink_ack, fields),
+      downlink_nack: pathExpand(template.downlink_nack, fields),
+      downlink_sent: pathExpand(template.downlink_sent, fields),
+      downlink_failed: pathExpand(template.downlink_failed, fields),
+      downlink_queued: pathExpand(template.downlink_queued, fields),
+      downlink_queue_invalidated: pathExpand(template.downlink_queue_invalidated, fields),
+      location_solved: pathExpand(template.location_solved, fields),
+      service_data: pathExpand(template.service_data, fields),
+    }
+
+    if (template.create_downlink_api_key) {
+      const key = {
+        name: `${webhook_id} downlink API key`,
+        rights: ['RIGHT_APPLICATION_TRAFFIC_DOWN_WRITE'],
+      }
+      const { key: downlink_api_key } = await createApplicationApiKey(appId, key)
+      webhook.downlink_api_key = downlink_api_key
+    }
+
+    return webhook
+  }
+
+  @bind
+  async existCheck(webhookId) {
+    const { appId, getWebhook } = this.props
+
+    try {
+      await getWebhook(appId, webhookId, [])
+      return true
+    } catch (error) {
+      if (isNotFoundError(error)) {
+        return false
+      }
+
+      throw error
+    }
+  }
+
+  @bind
+  async handleSubmit(webhook) {
+    const { appId, createWebhook } = this.props
+
+    await createWebhook(appId, webhook)
+  }
+
+  @bind
+  async handleSubmitSuccess() {
+    const { appId, navigateToList } = this.props
+
+    navigateToList(appId)
+  }
+
+  @bind
+  async handleWebhookSubmit(values, webhook, { setSubmitting, resetForm }) {
+    await this.setState({ error: '' })
```

Comment:
```suggestion
    this.setState({ error: undefined })
```

I think awaiting `setState` was never really doing anything.